api/database/migrations/2015_07_21_144551_create_articles_table.php

```diff
@@ -0,0 +1,37 @@
+<?php
+
+use Illuminate\Database\Schema\Blueprint;
+use Illuminate\Database\Migrations\Migration;
+
+class CreateArticlesTable extends Migration
+{
+    /**
+     * Run the migrations.
+     *
+     * @return void
+     */
+    public function up()
+    {
+        Schema::create(\App\Models\Article::getTableName(), function (Blueprint $table) {
+                $table->uuid('article_id')->primary();
+                $table->string('title', 255);
+                $table->text('content');
+                $table->string('permalink', 255)->index()->nullable();
+                $table->dateTime('first_published')->nullable();
+
+                $table->dateTime('created_at');
+                $table->dateTime('updated_at')->nullable();
+            }
+        );
+    }
+
+    /**
+     * Reverse the migrations.
+     *
+     * @return void
+     */
+    public function down()
+    {
+        Schema::drop(\App\Models\Article::getTableName());
+    }
+}
```

Comment:
Unless it is a nullable fk constraint. We can write the article.permalink to the database after the article is saved. Remember write speed is unimportant, so it is ok to be three writes in a transaction.
ie
1) article is created, but null article.permalink
2) permalink is created
3) article.permalink is written.